### PR TITLE
Compute CLI Commands: Features and bugs fixes

### PR DESCRIPTION
**Pull request by ruslangabitov**
Changes:
    Handle generation of SSH certs inside 'vm create' and 'vm docker create' commands.
    Added --lun option to 'vm disk attach-new' and 'vm disk attach' commands.
    Added functionality to set data disk name in 'vm disk attach-new' command

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@ruslangabitov__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: lib/util/certUtils._js [hunk 1]
```diff
@@ -209,3 +187,120 @@ exports.extractBase64X509CertFromPEM = function (x509CertPEM) {
...
+        } else {
+          return callback(null, false);
+        }
+      });
+    }
+  });
+};
+
+exports.generateSSHKeysIfNeeded = function (vmName, callback) {
+  var homePath = process.env[(process.platform == 'win32') ? 'USERPROFILE' : 'HOME'];
```
Comment (yugangw-msft) [inline lib/util/certUtils._js]: use "==="?

Comment (yugangw-msft) [inline lib/util/certUtils._js]: consider expose this one https://github.com/Azure/azure-xplat-cli/blob/dev/lib/util/utilsCore.js#L47

File: test/util/asmVMTestUtil.js [hunk 2]
```diff
@@ -513,7 +513,7 @@ asmVMTestUtil.prototype.deleteDockerCertificates = function(dockerCertDir, docke
         fs.unlinkSync(dockerCerts.client);
         fs.unlinkSync(dockerCerts.clientCert);
         fs.unlinkSync(dockerCerts.extfile);
-        fs.rmdirSync(dockerCertDir);
+        //fs.rmdirSync(dockerCertDir);
```
Comment (yugangw-msft) [inline test/util/asmVMTestUtil.js]: please put in comments for why avoid removing the folder

File: test/util/asmVMTestUtil.js [hunk 3]
```diff
@@ -563,6 +563,35 @@ asmVMTestUtil.prototype.checkForDockerCertificates = function(vmName, dockerCert
 
     return true;
 };
+asmVMTestUtil.prototype.checkForSSHKeys = function(vmName, SSHKeyDir) {
+	 sshKeys = {
```
Comment (yugangw-msft) [inline test/util/asmVMTestUtil.js]: again, please fix tab

Comment (yugangw-msft) [inline test/util/asmVMTestUtil.js]: also this are dupes to the same functions in other files, please consolidate

File: test/commands/arm/vm/arm.vm-create-generate-ssh-tests.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        vmPrefix = suite.isMocked ? vmPrefix : suite.generateId(vmPrefix, null);
+        nicName = suite.isMocked ? nicName : suite.generateId(nicName, null);
+        storageAccount = suite.generateId(storageAccount, null);
+        storageCont = suite.generateId(storageCont, null);
+        osdiskvhd = suite.isMocked ? osdiskvhd : suite.generateId(osdiskvhd, null);
+        vNetPrefix = suite.isMocked ? vNetPrefix : suite.generateId(vNetPrefix, null);
+        subnetName = suite.isMocked ? subnetName : suite.generateId(subnetName, null);
+        publicipName = suite.isMocked ? publicipName : suite.generateId(publicipName, null);
+        dnsPrefix = suite.isMocked ? dnsPrefix : suite.generateId(dnsPrefix, null);
+		homePath = process.env[(process.platform === 'win32') ? 'USERPROFILE' : 'HOME'];
```
Comment (yugangw-msft) [inline test/commands/arm/vm/arm.vm-create-generate-ssh-tests.js]: indent

File: test/commands/arm/vm/arm.vm-create-generate-ssh-tests.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    describe('vm', function() {
+
+      it('create with generate ssh keys option should pass', function(done) {
+        this.timeout(vmTest.timeoutLarge);
+        vmTest.checkImagefile(function() {
+          vmTest.createGroup(groupName, location, suite, function(result) {
+            if (VMTestUtil.linuxImageUrn === '' || VMTestUtil.linuxImageUrn === undefined) {
+              vmTest.GetLinuxSkusList(location, suite, function(result) {
+                vmTest.GetLinuxImageList(location, suite, function(result) {
+				  SSHKeyDir = path.join(homePath, SSHKeyFolder);
```
Comment (yugangw-msft) [inline test/commands/arm/vm/arm.vm-create-generate-ssh-tests.js]: please address all indents issues in all the test files

**Pull request merged**